notificationworkerlambda/src/main/scala/com/gu/notifications/worker/models/SendingResults.scala

```diff
@@ -20,5 +20,11 @@ object SendingResults {
     case Right(_) => previous.copy(successCount = previous.successCount + 1)
     case Left(_) => previous.copy(failureCount = previous.failureCount + 1)
   }
+
+  def aggregateBatch(previous: SendingResults, batchSize: Int, res: Either[Throwable, BatchDeliverySuccess]): SendingResults = res match {
+    case Right(batchSuccess) =>
+      batchSuccess.responses.foldLeft(previous)((acc, resp) => SendingResults.aggregate(acc, resp))
+    case Left(_) => SendingResults(0, batchSize, 0)
```

Comment:
I just wonder why the `previous` result is completely discarded when the current result is failure?

Reply:
Good catch - I'll update :)